templates/ra_guac_private_autoscale_elb.element

```diff
@@ -142,13 +148,22 @@
         "UseURLText2" :
         {
             "Fn::Not" : [ { "Fn::Equals" : [ { "Ref" : "URLText2" }, "" ] } ]
-        }
+        },
+        "UseAmiId" :
+        {
+            "Fn::Not" : [ { "Fn::Equals" : [ { "Ref" : "AmiId" }, "" ] } ]
+        },
+        "NotToUseAmiId" :
+        {
+              "Fn::Equals" : [ { "Ref" : "AmiId" }, "" ] 
+        }		
```

Comment:
Couldn't reverse the logic in the resource section but yea we can in the
ImageId section of the launch configuration.

That's a lot cleaner. I'll make those changes this morning.

On Sat, Jul 9, 2016 at 2:30 PM, Loren Gordon <notifications@github.com>
wrote:

> In templates/ra_guac_private_autoscale_elb.element
> <https://github.com/plus3it/cfn/pull/79#discussion_r70170480>:
>
> > @@ -142,13 +148,22 @@
> >          "UseURLText2" :
> >          {
> >              "Fn::Not" : [ { "Fn::Equals" : [ { "Ref" : "URLText2" }, "" ] } ]
> > -        }
> > +        },
> > +        "UseAmiId" :
> > +        {
> > +            "Fn::Not" : [ { "Fn::Equals" : [ { "Ref" : "AmiId" }, "" ] } ]
> > +        },
> > +        "NotToUseAmiId" :
> > +        {
> > +              "Fn::Equals" : [ { "Ref" : "AmiId" }, "" ]
> > +        }		
>
> Ahhh, ok, this *can* be collapsed to a single condition, just need to
> reverse the If logic. But first, rename NotToUseAmiID to UseAmiLookup. In
> the custom AmiIdLookup resource, nothing changes (except the renamed
> parameter). Then in the Fn::If statement, reverse the logic. E.g.
>
>                "ImageId" :
>                {
>                    "Fn::If" :
>                    [
>                            "UseAmiLookup",
>                            { "Fn::GetAtt" : [ "AmiIdLookup", "Id" ] },
>                            { "Ref" : "AmiId" }
>                    ]
>                },
>
> Take note of the whitespace above. Trying to standardize a bit. Braces and
> brackets should be on new lines, except where the entire object fits within
> a 79 char line, including the closing brace/bracket. Four space indents. No
> trailing whitespace at the end of a line. Single space before and after a
> colon, single space after opening brace/bracket, single space before
> closing brace/bracket (except when such characters are at the end of the
> line, no trailing whitespace takes precedent).
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/plus3it/cfn/pull/79/files/32b04593f1ea2f40393176b82d8a595a249a0efd#r70170480>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ANo0lymhMQhpJaiDA9XoKaK-v004YBBrks5qT-i1gaJpZM4JHXtN>
> .
>



-- 
Plus3 IT Systems - Cloud Engineer
304-279-4771
christopher.stotler@plus3it.com

-- 
 

------------------------------

*This e-mail and any attachments are intended only for the use of the 
addressee(s) named herein and may contain proprietary information. If you 
are not the intended recipient of this e-mail or believe that you received 
this email in error, please take immediate action to notify the sender of 
the apparent error by reply e-mail; permanently delete the e-mail and any 
attachments from your computer; and do not disseminate, distribute, use, or 
copy this message and any attachments.*